client/state/initial-state.js

```diff
@@ -87,16 +88,29 @@ export function persistOnChange( reduxStore, serializeState = serialize ) {
 	return reduxStore;
 }
 
+// This can't be a middleware as those are invoked before the reducer that
```

Comment:
Alternatively, I wonder if it's sufficient to call `next()` in a middleware, _then_ `getState()`. This probably doesn't handle asynchronous cases very well, but I'm having a hard time thinking of a need for asynchronous action creator title updating.

Have you considered creating a query component specifically for managing the document title. Not `<DocumentHead />` (since this isn't always used) but rather one that sits near the top of the `<Layout />` render hierarchy?